File: 1.0/src/cz/mg/vulkan/PFNvkGetDeviceGroupPeerMemoryFeaturesKHX.java
package cz.mg.vulkan;

public class PFNvkGetDeviceGroupPeerMemoryFeaturesKHX extends VkFunctionPointer {
    public PFNvkGetDeviceGroupPeerMemoryFeaturesKHX() {
    }

    public PFNvkGetDeviceGroupPeerMemoryFeaturesKHX(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkGetDeviceGroupPeerMemoryFeaturesKHX(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetDeviceGroupPeerMemoryFeaturesKHX(long value) {
        setValue(value);
    }

    public PFNvkGetDeviceGroupPeerMemoryFeaturesKHX(VkInstance instance) {
        super(instance, new VkString("vkGetDeviceGroupPeerMemoryFeaturesKHX"));
    }

    public void call(VkDevice device, VkUInt32 heapIndex, VkUInt32 localDeviceIndex, VkUInt32 remoteDeviceIndex, VkPeerMemoryFeatureFlagsKHX pPeerMemoryFeatures){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), heapIndex != null ? heapIndex.getVkAddress() : VkPointer.getNullAddressNative(), localDeviceIndex != null ? localDeviceIndex.getVkAddress() : VkPointer.getNullAddressNative(), remoteDeviceIndex != null ? remoteDeviceIndex.getVkAddress() : VkPointer.getNullAddressNative(), pPeerMemoryFeatures != null ? pPeerMemoryFeatures.getVkAddress() : VkPointer.NULL);
    }

    protected static native void call(long vkaddress, long device, long heapIndex, long localDeviceIndex, long remoteDeviceIndex, long pPeerMemoryFeatures);
}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkGetDeviceGroupPeerMemoryFeaturesKHX.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkGetDeviceGroupPeerMemoryFeaturesKHX_call(JNIEnv* env, jclass jc, jlong address, jlong device, jlong heapIndex, jlong localDeviceIndex, jlong remoteDeviceIndex, jlong pPeerMemoryFeatures){
    (void)env;
    (void)jc;
    PFN_vkGetDeviceGroupPeerMemoryFeaturesKHX f = (PFN_vkGetDeviceGroupPeerMemoryFeaturesKHX)jniLongToFunctionPointer(address);
    f(
        *((VkDevice*)jniLongToPointer(device)),
        *((uint32_t*)jniLongToPointer(heapIndex)),
        *((uint32_t*)jniLongToPointer(localDeviceIndex)),
        *((uint32_t*)jniLongToPointer(remoteDeviceIndex)),
        ((VkPeerMemoryFeatureFlagsKHX*)jniLongToPointer(pPeerMemoryFeatures))
    );
}
